phoenix-core-client/src/main/java/org/apache/phoenix/util/ViewUtil.java

```diff
@@ -484,7 +484,8 @@ public static void addIndexesFromParent(PhoenixConnection connection, PTable vie
                             .setTableName(modifiedIndexName)
                             .setViewStatement(viewStatement)
                             .setUpdateCacheFrequency(view.getUpdateCacheFrequency())
-                            .setTenantId(view.getTenantId())
+                            //retain the tenantId from the index being inherited
+                            .setTenantId(index.getTenantId())
```

Comment:
@palashc Can you add another test case where there are multiple tenant views on the same global view ? And what would happen if those tenant views have the same name (very common)  because they have a different tenant id ? I think they will now point to the same entry in the metadata cache.

Reply:
Added the test. 

> I think they will now point to the same entry in the metadata cache.

Yes. There was only one entry in the cache for the index with modified name and tenant id is null. Both tenant view PTables also have the index ptable with the modified name and no tenant id.